Update PR template (#2791)
model-checking/kani: .github/PULL_REQUEST_TEMPLATE.md

> Please ensure your PR description includes the following:

> > 1. Description on how your changes improve Kani.

Reviewer: ```suggestion
> 1. A description of how your changes improve Kani.
```

> > 2. Provide some context on the problem you are solving.

Reviewer: ```suggestion
> 2. Some context on the problem you are solving.
```

> > 3. List issues that are resolved by this PR.

Reviewer: ```suggestion
> 3. A list of issues that are resolved by this PR.
```

Reviewer 2: Do we even need this item? We already have `Resolves #ISSUE-NUMBER` below.

> ### Change Description

Reviewer 3: Can we please remove this header? Upon creating a PR, GitHub will place the commit message's body at the top. So the first manual step is to move this header.

Reviewer 4: I think the template should either not have any headers or use multiple to bring structure to the PR/the commit message. I don't think there would be anything wrong with having sections in the commit message if that makes it more readable.

> <!-- Describe how your changes improve Kani. Please provide some context on the problem you are solving. -->

Author: Done, I used a block quote instead.
